### DISCUSSION TITLE
SSH connection problems

### DISCUSSION THREAD

**Discussion by steelwing** [Security-Onion-Solutions/securityonion#7044] [Unsupported Versions] [answered]
Until today, I was unable to login remotely to my SecurityOnion server via SSH.  I could access the box via the web console, but SSH kept giving me errors about "no matching host key type found."
Digging through SSHD_config, I found 4 items at the end, defining ciphers, macs, kexalgorithms, and hostkeyalgorithms.  I commented out these entries and put in just ciphers and macs that were compliant with FIPS 140-2.  I then restarted SSHD and it connected without a problem.
What do kexalgorithms and hostkeyalgorithms do?  Is there a SecurityOnion-related reason to define those?  Does part of SecurityOnion break if you take those out of sshd_config?

**Comment by dougburks** [comment, marked as answer]
https://docs.securityonion.net/en/2.3/ssh.html#hardening

**Reply by steelwing** [reply]
Thank you for your help!
